autogen/agentchat/conversable_agent.py

```diff
@@ -1988,3 +2005,28 @@ def get_total_usage(self) -> Union[None, Dict[str, int]]:
             return None
         else:
             return self.client.total_usage_summary
+
+
+def register_function(
```

Comment:
Shall we create an alias `register_tool`?

Reply:
And export it to make it top-level? i.e., `from autogen import register_function`

Reply:
I think this would be wrong in the long run. Suppose OpenAI adds another tool to their API, and it is reasonable to assume they will do it, a that we have to support it as well. Say this tool is for calling a REST endpoint. Then we would need to add a function `register_tool_rest` and we would rename `register_tool` to `register_tool_function`. But there is no need to have `tool` in the name from the user perspective, they just want to register their functions and REST endpoints and they don't care what exactly is JSON structure for declaring them to OpenAI API. So we could simplify names to `register_rest` and `register_function`.

Reply:
I see. makes sense.